.gitmodules

```diff
@@ -1,3 +1,4 @@
 [submodule "test/data/quicknxs-data"]
 	path = test/data/quicknxs-data
 	url = https://code.ornl.gov/sns-hfir-scse/infrastructure/test-data/quicknxs-data.git
+	branch = main
```

Comment:
I'm confused by this line. What is the behavior of `git submodule update` when the parent repo stores a SHA to the submodule repo that is not the tip of  `main`?